### PR TITLE
[TASK-21] As a user, I can navigate through survey questions

### PR DESCRIPTION
**Pull request by rafayet-monon**
Task Card - https://github.com/rafayet-monon/surveyor/projects/1#card-45376645

## What happened
Navigate questions of a survey.
 
## Insight
- [x] Connect backend API and build appropriately for easier use
- [x] Show question components based on the question type
- [x] Show questions based on API

## Proof Of Work
React App

### CODE REVIEW

File: src/stylesheets/mixins/next-icon-container.scss [hunk 1]
```diff
@@ -0,0 +1,17 @@
+@mixin next-icon-container() {
+  padding: rem(14px 13px);
+  position: relative;
```
Comment (rafayetn) [inline src/stylesheets/mixins/next-icon-container.scss:2]: - :warning: 
	  Expected "position" to come before "padding" in group "Positioning" (order/properties-order)
- :warning: Expected an empty line before property "position" (order/properties-order)

File: src/contexts/details.js [hunk 2]
```diff
@@ -2,9 +2,13 @@ import React, { createContext } from 'react';
...
 const DetailsContext = createContext();
 
-const DetailsProvider = ({ surveyDetail, children }) => {
+const DetailsProvider = ({ surveyDetail, questionList, children }) => {
   return (
-    <DetailsContext.Provider value={ surveyDetail }>
+    // Using the prebuild questionList and passing it to context provider
+    // so that it can be used among its' children
+    <DetailsContext.Provider
+      value={{ surveyDetail: surveyDetail, questionList: questionList }}
```
Comment (carryall) [inline src/contexts/details.js]: How about like this?
```suggestion
      value={{ surveyDetail, questionList }}
```

File: src/screens/SurveyDetail/QuestionBuilders/buildQuestionList.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+  questions.forEach((question, index) => {
+    const question_details = included.find(
+      (details) => details.id === question.id
+    );
+
+    questions[index]['type'] = question_details.attributes.display_type;
+    questions[index]['pick'] = question_details.attributes.pick;
+    questions[index]['text'] = question_details.attributes.text;
+    questions[index]['answers'] = question_details.relationships.answers.data;
```
Comment (carryall) [inline src/screens/SurveyDetail/QuestionBuilders/buildQuestionList.js]: Not sure why we're not using `question` here 🤔 

```suggestion
    question['type'] = question_details.attributes.display_type;
    question['pick'] = question_details.attributes.pick;
    question['text'] = question_details.attributes.text;
    question['answers'] = question_details.relationships.answers.data;
```

Comment (rafayet-monon) [reply src/screens/SurveyDetail/QuestionBuilders/buildQuestionList.js]: @carryall fixed reviews in [c300c8b](https://github.com/rafayet-monon/surveyor/pull/28/commits/c300c8b52f96bcfbb3a8ebe9eb92f84ee39b2c9e)

File: src/screens/SurveyDetail/question.js [hunk 4]
```diff
@@ -1,22 +1,28 @@
...
-    { name: 'India', value: 'IN' },
-    { name: 'Nepal', value: 'NP' },
-    { name: 'Bhutan', value: 'BH' },
-    { name: 'Thailand', value: 'TH' },
-    { name: 'Singapore', value: 'SG' },
-    { name: 'Vietnam', value: 'VN' }
-  ];
+  const detailsContext = useContext(DetailsContext);
+  const filteredQuestions = filterQuestionList(detailsContext.questionList);
+  const [currentQuestionProperties, setCurrentQuestionProperties] = useState(
```
Comment (carryall) [inline src/screens/SurveyDetail/question.js]: How about renaming this so it become shorter 😄 
```suggestion
  const [currentQuestion, setCurrentQuestion] = useState(
```

File: src/tests/screens/SurveyDetail/detail.test.js [hunk 5]
```diff
@@ -8,14 +8,22 @@ import { Router } from 'react-router-dom';
...
+
   it('shows the survey image', () => {
     const history = createMemoryHistory();
     const { container } = render(
       <Router history={ history }>
-        <DetailsContext.Provider value={ SurveyDetailResponse.data }>
+        <DetailsContext.Provider
+          value={{
+            surveyDetail: SurveyDetailResponse,
+            questionList: questionList
```
Comment (carryall) [inline src/tests/screens/SurveyDetail/detail.test.js]: How about added the [object-shorthand](https://eslint.org/docs/rules/object-shorthand) rule to ESlint too? 💭 
```suggestion
            questionList
```

Comment (rafayet-monon) [reply src/tests/screens/SurveyDetail/detail.test.js]: @carryall fixed reviews in [b93b1a3](https://github.com/rafayet-monon/surveyor/pull/28/commits/b93b1a3c57c4fd3bc24715e1093d6ef291fbbe7a)

File: src/components/WheelSelect/index.js [hunk 6]
```diff
@@ -4,7 +4,20 @@ import Slider from 'react-slick';
 
 import slickSettings from 'components/WheelSelect/slickSettings';
 
-const WheelSelect = ({ data, multipleSelect }) => {
+const WheelSelect = ({ options, multipleSelect }) => {
+  // Adding some demo data for now to show components based on question type.
+  const demoData = [
+    {
+      display: 'N0',
```
Comment (carryall) [inline src/components/WheelSelect/index.js]: @rafayet-monon I just realized this `NO` not `No` but it's okay since we will remove it in another PR 😅

**Review by carryall — APPROVED** [review]

File: src/screens/SurveyDetail/QuestionBuilders/buildQuestionList.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
+// Build question list and answers from the api for easier of question and
```
Comment (carryall) [inline src/screens/SurveyDetail/QuestionBuilders/buildQuestionList.js]: @rafayet-monon Please correct me if I'm wrong, as I understand that 
- `determineQuestionType` render a question according to the question type
- `buildQuestionList` function build the question list from the API response
- `filterQuestionList` function filter questions out from the API response
- `questionProperties` function prepare property for the question to make it ready to render

so IMO we can
- move `buildQuestionList` and `filterQuestionList` to helpers
- make `determineQuestionType` a separate component
- `questionProperties` can either be a helper function or a private function in question component

what do you think?

Comment (rafayet-monon) [reply src/screens/SurveyDetail/QuestionBuilders/buildQuestionList.js]: @carryall Fixed your concerns in [25c0413](https://github.com/rafayet-monon/surveyor/pull/28/commits/25c0413341e4022d58b56479fe2520743df8fb71)

**Review by carryall — APPROVED** [review]

**Pull request merged**